File: LCM.py
# Least Common Multiple (LCM)
# - Compute the LCM of two numbers using GCD.
# - Time: O(log(min(a,b))), Space: O(1)

class Solution:
    def lcm(self, a: int, b: int) -> int:
        from math import gcd
        return abs(a * b) // gcd(a, b)

File: LCM.cs
// Least Common Multiple (LCM)
// - Compute the LCM of two numbers using GCD.

using System;

class Solution
{
    public int LCM(int a, int b)
    {
        return Math.Abs(a * b) / GCD(a, b);
    }

    private int GCD(int a, int b)
    {
        while (b != 0)
        {
            int temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }
}
